pkg/controllers/allocation/suite_test.go

```diff
@@ -12,17 +12,19 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package allocation
+package allocation_test
```

Comment:
Why change up the package here? I know it's a testing style, but it's outside of the convention of the project thus far. Should we be moving to adopt this pattern? I find it a bit of a surprising convention of golang and I haven't yet seen the value over just using the same package.

Reply:
it's a protection against using non-public fields. For example, when I changed the AllocationController to have public fields so we can configure the batcher in the tests, it caught it in it's own suite test which was more noticeable since I was working in that package.  I actually thought about changing it back without the "_test", but the cloudprovider suite tests also needs to create a controller, so it really should behave the same. 

In general, I think we should absolutely stick with this convention.